apiserver/facades/controller/charmrevisionupdater/register.go

```diff
@@ -35,14 +35,27 @@ func newCharmRevisionUpdaterAPI(ctx facade.ModelContext) (*CharmRevisionUpdaterA
 			err,
 		)
 	}
-	newCharmhubClient := func(st State) (CharmhubRefreshClient, error) {
+	modelConfigService := ctx.ServiceFactory().Config()
+
+	newCharmhubClient := func(stdCtx context.Context) (CharmhubRefreshClient, error) {
```

Comment:
Note for @SimonRichardson, it would be nice to be able to refactor all of this into a model charm service Simon so we are asking specific question on that service and can ditch model config usage here.

Reply:
I don't think that should be a charm service, that potentially should be a charmhub service. I'm not sure it's a charm concern how you get a charm. Otherwise, everything is going to be in the application domain.

Also, I don't think we should go on this quest, as you then need to push the HTTP client through the service factory so you can create it. It is a dependency after all. For now, let's just make a mental note and leave it as is.